File: 0049-group-anagrams/solution.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> m = new HashMap<>();
        for (String s : strs) {
            char[] ch = s.toCharArray();
            Arrays.sort(ch);
            String str = new String(ch);
            if (!m.containsKey(str)) {
                m.put(str, new ArrayList<>());
            }
            m.get(str).add(s);
        }
        return new ArrayList<>(m.values());
    }
}

File: 0049-group-anagrams/solution.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<string>> m;

        for(auto s : strs){
            string word = s;
            sort(word.begin(),word.end());
            m[word].push_back(s); 
        }

        vector<vector<string>> result;
        for(auto x : m){
            result.push_back(x.second);
        }
        return result;
        
    }
};
